llvm/utils/update_failed_tests.py

```diff
@@ -0,0 +1,110 @@
+#!/usr/bin/env python3
+
+"""A test case update orchestrator script.
+
+This script is a utility to automate the updating of failed LLVM test cases by
+invoking other `update_*_test_checkes.py` scripts. It provides a one-click
+solution to update test expectations for previously failed tests.
+"""
+
+import subprocess
+import os
+
+
+def get_llvm_path():
+    util_path = os.path.dirname(os.path.realpath(__file__))
+    llvm_path = os.path.dirname(util_path)
+    print("The LLVM path is", llvm_path)
+    return llvm_path
+
+
+def get_build_path():
+    if os.path.basename(os.getcwd()).startswith("build"):
+        build_path = os.getcwd()
+    else:
+        dirs = [d for d in os.listdir('.') if os.path.isdir(d)]
+        build_dirs = [d for d in dirs if d.startswith('build')]
+
+        if len(build_dirs) != 1:
+            print(
+                "Cannot find the build directory. Please run this script in the build directory.")
+            exit(1)
+        build_path = build_dirs[0]
+
+    print("The BUILD path is", build_path)
+    return build_path
```

Comment:
If we want to add this as a separate script and not part of lit, we should be generating the script (or at least a wrapper around it) in the build directory.